engine/tx_broadcast.go

```diff
@@ -4,43 +4,31 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/bitcoin-sv/spv-wallet/engine/chainstate"
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
 	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
 )
 
-func broadcastTransaction(ctx context.Context, tx *Transaction) error {
-	client := tx.Client()
-	chainstateSrv := client.Chainstate()
+func broadcastTransaction(ctx context.Context, txModel *Transaction) error {
+	client := txModel.Client()
 
-	defer recoverAndLog(tx.Client().Logger())
+	defer recoverAndLog(txModel.Client().Logger())
 
 	unlock, err := newWriteLock(
```

Comment:
Minor, but 
Why do we actually need this lock here 🤔

Reply:
Don't know exactly. I believe we should not move it to out "new transaction-flow".
For now, I would not touch this 🤞